hibernate-core/src/main/java/org/hibernate/cfg/Ejb3Column.java

```diff
@@ -240,15 +240,17 @@ protected void initMappingColumn(
 			this.mappingColumn.setSqlType( sqlType );
 			this.mappingColumn.setUnique( unique );
 
-			if(writeExpression != null && !writeExpression.matches("[^?]*\\?[^?]*")) {
+			if ( writeExpression != null && !writeExpression.matches("[^?]*\\?[^?]*" ) ) {
```

Comment:
There is a missing space char right after `.matches(`. The expression seems nontrivial. Will the above statement incur repeated performance cost to compile the regular expression?

Reply:
However, I went a different approach and simply iterate the char values of the String counting the number of `?` chars by adding `StringHelper#count` which is similar to `StringHelper#countUnquoted`